src/reporter/coverage-report.xsl

```diff
@@ -1,407 +1,436 @@
+
+<pkg:import-uri>http://www.jenitennison.com/xslt/xspec/coverage-report.xsl</pkg:import-uri>
+
+<xsl:param name="inline-css" as="xs:string" select="false() cast as xs:string" />
+
+<xsl:param name="report-css-uri" as="xs:string?" />
+
+<!-- @use-character-maps for inline CSS -->
+<xsl:output method="xhtml" use-character-maps="test:disable-escaping" />
+
+<xsl:variable name="trace" as="document-node()" select="/" />
+
+<xsl:variable name="xspec-uri" as="xs:anyURI" select="$trace/trace/@xspec" />
+<xsl:variable name="xspec-doc" as="document-node(element(x:description))"
+  select="doc($xspec-uri)" />
+
+<xsl:variable name="stylesheet-uri" as="xs:anyURI"
+  select="resolve-uri($xspec-doc/x:description/@stylesheet, $xspec-uri)" />
+
+<xsl:variable name="stylesheet-trees" as="document-node()+"
+  select="test:collect-stylesheets(doc($stylesheet-uri))" />
+
+<xsl:function name="test:collect-stylesheets" as="document-node()+">
+  <xsl:param name="stylesheets" as="document-node()+" />
+  <xsl:variable name="imports" as="document-node()*"
+    select="document($stylesheets/*/(xsl:import|xsl:include)/@href)" />
+  <xsl:variable name="new-stylesheets" as="document-node()*"
+    select="$stylesheets | $imports" />
+  <xsl:choose>
+    <xsl:when test="$imports except $stylesheets">
+      <xsl:sequence select="test:collect-stylesheets($stylesheets | $imports)" />
+    </xsl:when>
+    <xsl:otherwise>
+      <xsl:sequence select="$stylesheets" />
+    </xsl:otherwise>
+  </xsl:choose>
+</xsl:function>
+
+<xsl:key name="modules" match="m" use="@u" />
+<xsl:key name="constructs" match="c" use="@id" />
+<xsl:key name="coverage" match="h" use="@m || ':' || @l" />
+
+<xsl:template match="/">
+  <xsl:apply-templates select="." mode="test:coverage-report" />
+</xsl:template>
+
+<xsl:template match="/" mode="test:coverage-report">
+  <html>
+    <head>
+      <title>
+        <xsl:text expand-text="yes">Test Coverage Report for {x:format-uri($stylesheet-uri)}</xsl:text>
+      </title>
+      <xsl:call-template name="test:load-css">
+        <xsl:with-param name="inline" select="$inline-css cast as xs:boolean" />
+        <xsl:with-param name="uri" select="$report-css-uri" />
+      </xsl:call-template>
+    </head>
+    <body>
+      <h1>Test Coverage Report</h1>
+      <p>
+        <xsl:text>Stylesheet:  </xsl:text>
+        <a href="{$stylesheet-uri}">
+          <xsl:value-of select="x:format-uri($stylesheet-uri)" />
+        </a>
+      </p>
+      <xsl:apply-templates select="$stylesheet-trees/xsl:*" mode="test:coverage-report" />
+    </body>
+  </html>
+</xsl:template>
+  
+<xsl:template name="body-content">
+  <xsl:param name="number-of-lines" as="xs:integer"/>
+  <xsl:param name="module" as="xs:string" tunnel="yes"/>
+  <xsl:param name="stylesheet-string" as="xs:string"/>
+  <xsl:param name="number-format" as="xs:string" tunnel="yes"/>
+  <h2>
+    <xsl:text expand-text="yes">module: {x:format-uri($stylesheet-uri)}; {$number-of-lines} lines</xsl:text>
```

Comment:
In case of imported stylesheets, `$stylesheet-uri` at this line is incorrect. For example, [`tutorial\coverage\demo.xspec`](https://github.com/xspec/xspec/blob/master/tutorial/coverage/demo.xspec).

Reply:
This is still true.